SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialGDKSpanId.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Interop/Connection/SpatialGDKSpanId.h"
+
+// ----- FSpatialGDKSpanId -----
+
+FSpatialGDKSpanId::FSpatialGDKSpanId()
+	: bIsValid(false)
+{
+}
+
+FSpatialGDKSpanId::FSpatialGDKSpanId(bool bInIsValid)
+	: bIsValid(bInIsValid)
+{
+	if (bIsValid)
+	{
+		WriteData(Trace_SpanId_Null());
+	}
+}
+
+FSpatialGDKSpanId::FSpatialGDKSpanId(const Trace_SpanIdType* TraceSpanId)
+	: bIsValid(true)
+{
+	WriteData(TraceSpanId);
+}
+
+void FSpatialGDKSpanId::WriteData(const Trace_SpanIdType* TraceSpanId)
+{
+	const int32 Size = TRACE_SPAN_ID_SIZE_BYTES * sizeof(Trace_SpanIdType);
```

Comment:
This looks wrong btw, Size = num_bytes * sizeof(pointer) (which is 8 on x64) which is probably not what you intended. The memcpy should probably just be the macro size

Reply:
Trace_SpanIdType is just uint8 but you are right, we should just use TRACE_SPAN_ID_SIZE_BYTES